src/firewall/functions.py

```diff
@@ -26,6 +33,41 @@
 }
 
 
+def str_to_bool(val, default=DEFAULT_SENTINEL, ignore_error=False):
+
+    is_default = False
+
+    if isinstance(val, str):
+        val2 = val.lower().strip()
+        if val2 in ("1", "y", "yes", "true", "on"):
+            return True
+        if val2 in ("0", "n", "no", "false", "off"):
+            return False
+        if val2 in ("", "default", "-1"):
+            is_default = True
+    elif val is None:
+        # None is (maybe) accepted as default value.
+        is_default = True
+    elif isinstance(val, bool):
+        # For convenience, also accept that the value is already a boolean.
+        return val
+
+    if is_default:
+        if default is DEFAULT_SENTINEL:
+            # No default is specified. We don't allow a default value.
+            pass
```

Comment:
We should always have a default value. This should not be necessary.

Reply:
as explained, that is wrong.